osc-server/src/main/java/org/osc/core/broker/util/network/NetworkSettingsApi.java

```diff
@@ -74,71 +49,36 @@ public NetworkSettingsDto getNetworkSettings() {
         if (ServerUtil.isWindows()) {
             return networkSettingsDto;
         }
-
-        Properties networkInterfaceConfig = new Properties();
-
+        networkSettingsDto.setDhcp(true);
+        String hostIpAddress = null;
         try {
-            networkInterfaceConfig = FileUtil.loadProperties(this.DEFAULT_INTERFACE_CONFIG_FILE);
-        } catch (IOException e) {
-            log.error("Failed to load network settings", e);
-        }
-
-        String bootProtocol = networkInterfaceConfig.getProperty(this.BOOTPROTOCOL, "");
-
-        // return IP, netmask, GW, and DNS IP in static mode
-        if (bootProtocol.equals("static")) {
-
-            networkSettingsDto.setDhcp(false);
-            networkSettingsDto.setHostIpAddress(networkInterfaceConfig.getProperty(this.IPADDRESS));
-            networkSettingsDto.setHostSubnetMask(networkInterfaceConfig.getProperty(this.NETMASK));
-
-            Properties networkConfig = new Properties();
-            try {
-                networkConfig = FileUtil.loadProperties(this.NETWORK_CONFIG_FILE);
-            } catch (IOException e) {
-                log.error("Failed to load network settings", e);
-            }
-
-            networkSettingsDto.setHostDefaultGateway(networkConfig.getProperty(this.GATEWAY));
-            try {
-                String[] dns = getDNSSettings();
-                networkSettingsDto.setHostDnsServer1(dns[0]);
-                networkSettingsDto.setHostDnsServer2(dns[1]);
-            } catch (IOException e) {
-                log.error("Failed to load DNS settings", e);
-            }
-
-        } else {
-            // only return IP in DHCP mode
-            networkSettingsDto.setDhcp(true);
-            String hostIpAddress = null;
-            try {
-                hostIpAddress = NetworkUtil.getHostIpAddress();
-            } catch (SocketException | UnknownHostException e) {
-                log.error("Failed to get host and/or ip address", e);
-            }
+            hostIpAddress = NetworkUtil.getHostIpAddress();
             networkSettingsDto.setHostIpAddress(hostIpAddress);
-            networkSettingsDto.setHostSubnetMask("");
-            networkSettingsDto.setHostDefaultGateway("");
-            networkSettingsDto.setHostDnsServer1("");
-            networkSettingsDto.setHostDnsServer2("");
-        }
-
+            networkSettingsDto.setHostSubnetMask(getNetMask(hostIpAddress));
```

Comment:
What are the values returned in these fields for a typical OSC container deployment?

Reply:
In the Case of DHCP ,it  returns only IP address. Others are null value.